File: src/asp_net_backend/src/api/Models/Room.cs
namespace Restify.API.Models;

public class Room
{
	public uint	LodgingId { get; set; }
	public uint	Number { get; set; }
	public uint	TypeId { get; set; }

	public Lodging	Lodging { get; }
	public RoomType	Type { get; }
	
	public Room() {}

	public Room(RoomType type)
	{
		 Type = type;
	}
}

File: src/frontend/src/app/models/room.ts
import { Lodging } from "./lodging";
import { RoomType } from "./room_type";

export class Room
{
    public constructor(
        public lodgingId:   number,
        public number:  number,
        public typeId:      number,
        public lodging:     Lodging | null,
        public type:        RoomType | null
    ) { }
}